docs/kratos/concepts/email-sms.md

```diff
@@ -133,6 +134,112 @@ courier:
       # the configuration structure is the same as the verification
 ```
 
+### Custom Headers
+
+You can configure custom SMTP headers. For example, if integrating with AWS SES
+SMTP interface, the headers can be configured for cross-account sending:
+
+```yaml title="path/to/my/kratos/config.yml"
+# $ kratos -c path/to/my/kratos/config.yml serve
+courier:
+  smtp:
+    headers:
+      X-SES-SOURCE-ARN: arn:aws:ses:us-west-2:123456789012:identity/example.com
+      X-SES-FROM-ARN: arn:aws:ses:us-west-2:123456789012:identity/example.com
+      X-SES-RETURN-PATH-ARN: arn:aws:ses:us-west-2:123456789012:identity/example.com
+```
+
+## Sending SMS
+
+For sending SMS Ory Kratos uses an external SMS gateway, which must have HTTP
+API (such as Twilio, your local SMS sender, or your own microservice). Request
+method, headers, body, and content-type are fully configurable using options
+below.
+
+### Configuration
+
+Default configuration of Ory Kratos does not include sending SMS. To enable it
+you need to set "enabled" flag to true, sender URL, authorization (if needed)
+and request body format.
+
+#### Sender phone number
+
+SMS message will come to the recipient from this phone number. Depending on your
+SMS sender settings you can use letters here (for example "Your Org Name").
+Default value is equal to "Ory Kratos".
+
+```yaml title="path/to/my/kratos/config.yml"
+# $ kratos -c path/to/my/kratos/config.yml serve
+courier:
+  sms:
+    from: '+12065550110'
+```
+
+#### Request configuration
+
+```yaml title="path/to/my/kratos/config.yml"
+# $ kratos -c path/to/my/kratos/config.yml serve
+courier:
+  sms:
```

Comment:
Since this is so generic now, would it make sense to just call it the HTTP-RPC courier? And that can be used for sending SMS, but could also be wired to e.g. some CRM?

Or is there still some magic that happens that is required for SMS?

Reply:
Yes, there are templates defined for SMS, so this integration can't be reused for more generic request sending. I think that it would be better to create a dedicated method in courier that will allow sending just plain HTTP requests